### ISSUE TITLE
Nonexistent transitive dependency

### ISSUE THREAD

**Issue by MHajoha** [java-json-tools/jackson-coreutils#58]
When using `jackson-coreutils:2.0` in a Maven project, `jackson-coreutils:2.0-SNAPSHOT` is transitively required which does not exist in Maven Central or any other repository known to [mvnrepository.com](https://mvnrepository.com/artifact/com.github.java-json-tools/jackson-coreutils).

**Reproduction**
1. Create an otherwise empty project with a minimal POM depending on `jackson-coreutils:2.0` ([Gist](https://gist.github.com/MHajoha/0f36b2608d9445080594f81358bc9b66))
2. Try to build said project

**Expected Behaviour**
It builds.

**Actual Behaviour**
It does not build.
```
[ERROR] Failed to execute goal on project jackson-coreutils-dependency-example: Could not resolve dependencies for project org.example:jackson-coreutils-dependency-example:jar:1.0-SNAPSHOT: Failed to collect dependencies at com.github.java-json-tools:jackson-coreutils:jar:2.0 -> com.github.java-json-tools.jackson-coreutils:jackson-coreutils-equivalence:jar:2.0 -> com.github.java-json-tools:jackson-coreutils:jar:2.0-SNAPSHOT
```

The missing dependency needs to be explicitly excluded wherever it is used.

**Comment by Capstan** [comment]
The [pom for `jackson-coreutils-equivalence` 1.0](https://repo1.maven.org/maven2/com/github/java-json-tools/jackson-coreutils-equivalence/1.0/jackson-coreutils-equivalence-1.0.pom) does not have a dependency on `2.0-SNAPSHOT` of `jackson-coreutils`. There is no published v2 of `jackson-coreutils-equivalence`.

The [pom for `jackson-coreutils` 2.0](https://repo1.maven.org/maven2/com/github/java-json-tools/jackson-coreutils/2.0/jackson-coreutils-2.0.pom) does not refer to `jackson-coreutils-equivalence` at all.

So, I've not reproduced your gist, but I'm not understanding how this would transpire. Are you perhaps importing an older pre-release `2.0-SNAPSHOT` build? I could imagine those not all being brought up to date before release.

**Comment by Capstan** [comment]
In general, no release build should depend on any SNAPSHOT build, so if we do find such a thing, it should be fixed and re-released.

**Comment by MHajoha** [comment]
I did some more digging and it seems the broken POMs I pulled in come from jitpack.io:
* https://jitpack.io/com/github/java-json-tools/jackson-coreutils/2.0/jackson-coreutils-2.0.pom
* https://jitpack.io/com/github/java-json-tools/jackson-coreutils/jackson-coreutils-equivalence/2.0/jackson-coreutils-equivalence-2.0.pom

It's one the the repositories our own Artifactory caches and it transparently builds and serves artifacts from GH repos.
Appearantly it built a jackson-coreutils:2.0 on 2020-05-28 and has been serving that artifact ever since: https://jitpack.io/com/github/java-json-tools/jackson-coreutils/2.0/build.log

I'm guessing that build took place before 549007e035a878b9cb5cc7f51beeca56817c731c. (It's dated before the build but perhaps it wasn't pushed immediately?)

Anyway, thanks for your help. I'm going to look into why we pull from jitpack.io in the first place and whether we can stop.
